src/test/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepositoryTest.java

```diff
@@ -0,0 +1,226 @@
+
+    @Mock
+    private ApplicationParams applicationParams;
+
+    @Mock
+    private SecurityUtils securityUtils;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private ResponseEntity<RoleAssignmentResponse> responseEntity;
+
+    @InjectMocks
+    DefaultRoleAssignmentRepository defaultRoleAssignmentRepository;
+
+    private static final String ID = "4d96923f-891a-4cb1-863e-9bec44d1689d";
+    private static final String ACTOR_ID = "567567";
+    private static final String ROLE_TYPE = RoleType.ORGANISATION.name();
+    private static final String ROLE_NAME = "judge";
+
+    private static final String ATTRIBUTES_CONTRACT_TYPE = "SALARIED";
+    private static final String ATTRIBUTES_CASE_ID = "1504259907353529";
+
+    private static final String AM_URI = "/am/role-assignments/query";
+
+    private static final String TEST_PAGE_SIZE = "1";
+    private static final String TEST_TOTAL_RECORDS = "2";
+
+    private static final String HTTP_400_ERROR_MESSAGE = "Client error when getting Role Assignments from "
+        + "Role Assignment Service because of 400 BAD_REQUEST";
+    private static final String HTTP_404_ERROR_MESSAGE = "No Role Assignments found for userIds=[%s] and "
+        + "casesIds=[%s] when getting from Role Assignment Service because of 404 NOT_FOUND";
+    private static final String HTTP_500_ERROR_MESSAGE = "Problem getting Role Assignments from Role Assignment "
+        + "Service because of 500 INTERNAL_SERVER_ERROR";
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        defaultRoleAssignmentRepository = new DefaultRoleAssignmentRepository(
+            applicationParams, securityUtils, restTemplate);
+
+        when(applicationParams.getRoleAssignmentPageSize()).thenReturn(TEST_PAGE_SIZE);
+        when(applicationParams.amQueryRoleAssignmentsURL()).thenReturn(AM_URI);
+
+        when(securityUtils.authorizationHeaders()).thenReturn(new HttpHeaders());
+
+        responseEntity = ResponseEntity.ok().headers(responseHeaders()).body(responseBody());
+        doReturn(responseEntity).when(restTemplate).exchange(eq(AM_URI), eq(POST),
+            any(), eq(RoleAssignmentResponse.class));
+    }
+
+    @DisplayName("should return roleAssignments by user and roles")
+    @Test
+    void shouldReturnRoleAssignmentsByUserAndRoles() {
+        when(applicationParams.isRoleAssignmentPaginationEnabled()).thenReturn(false);
+
+        RoleAssignmentResponse roleAssignmentResponse = defaultRoleAssignmentRepository
+            .findRoleAssignmentsByCasesAndUsers(Collections.singletonList(ATTRIBUTES_CASE_ID),
+                Collections.singletonList(ACTOR_ID));
+
+        assertThat(roleAssignmentResponse.getRoleAssignments().size(), is(1));
+    }
+
+    @DisplayName("should error on 404 when post FindRoleAssignmentsByCasesAndUsers")
+    @Test
+    void shouldErrorOn404WhenPostFindRoleAssignmentsByCasesAndUsers() {
```

Comment:
http-404 from upstream should not happen, as empty role assignment list should come with http-200. Therefore, http-404 can only happen due to an internal implementation mistake, if it's not a false-positive from downstream. So, this needs to be translated as 500 to upstream.

Reply:
Existing behaviour. Can be done as part of Kaan's PR or in a new followup tech debt ticket?